### DISCUSSION TITLE
Way to opt out of startTransition conditionally?

### DISCUSSION THREAD

**Discussion by XiNiHa** [remix-run/react-router#10940] [Proposals]
I'm aware that `v7_startTransition` is behind a future flag, and I understand that unsetting the flag should solve my issue. However, I generally want the behavior of the flag between most of the page transitions (since that provides a smoother experience), while being able to conditionally opt out of it (to make `useRevalidator().revalidate()` trigger fallbacks in my case). Do you have any ideas to achieve this? At least adding an option to `.revalidate()` for disabling the transition would solve my usecase, but I'm not sure that's widely adoptable across all router state operations.

**Comment by einarq** [comment]
@XiNiHa 
Maybe this helps you?

https://github.com/remix-run/react-router/discussions/12855#discussioncomment-12684796

**Reply by XiNiHa** [reply]
Looks like it could be helpful in some cases! Though not all due to nature of how `flushSync` works 🥲
